main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -1439,11 +1439,22 @@ object Typer {
           resultEff = valueEff
         } yield (valueConstrs, resultTyp, resultPur, resultEff)
 
-      case KindedAst.Expression.NewObject(clazz, _, loc) =>
-        val resultTyp = getFlixType(clazz)
-        val resultPur = Type.Impure
-        val resultEff = Type.Empty
-        liftM(List.empty, resultTyp, resultPur, resultEff)
+      case KindedAst.Expression.NewObject(clazz, methods, loc) => {
+        def visitJvmMethod(method: KindedAst.JvmMethod): InferMonad[(List[Ast.TypeConstraint], Type, Type, Type)] = method match {
+          case KindedAst.JvmMethod(ident, fparams, exp, returnTpe, pur, eff, loc) =>
```

Comment:
@mlutze Do we need to do something here to ensure the formal parameter's type matches or do we get it for free?

Reply:
You're right, we need to check that they match the fparams from the class.

(Probably should also have a test for this :) )